dag_in_context/src/utility/canonicalize.egg

```diff
@@ -56,36 +59,29 @@
 
 (rewrite (SubTuple expr x 0)
          (Empty ty ctx)
+         :subsume
          :when ((HasArgType expr ty) (ContextOf expr ctx))
          :ruleset always-run)
 
 (rewrite (SubTuple expr x 1)
          (Single (Get expr x))
+         :subsume
          :ruleset always-run)
 
 (rewrite (SubTuple expr a b)
          (Concat (Single (Get expr a)) (SubTuple expr (+ a 1) (- b 1)))
+         :subsume
          :when ((> b 1))
          :ruleset always-run)
 
-; Also figure out what existing expressions are subtuples of other things
-; this helps remove concat layers
-(rule ((Get expr i))
-      ((union (Single (Get expr i))
-              (SubTuple expr i 1)))
+; ;; This rule makes sure every tuple can be 
```

Comment:
this is actually unnecessary because we are already generating gets for every "important" tuple. Removed in my new PRs.